src/Microsoft.Health.Dicom.Core/Features/Validation/TagPathValidator.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Text.RegularExpressions;
+using Microsoft.Health.Dicom.Core.Exceptions;
+
+namespace Microsoft.Health.Dicom.Core.Features.Validation
+{
+    public static class TagPathValidator
+    {
+        public static void Validate(string tagPath)
+        {
+            Match match = Regex.Match(tagPath, @"\A\d{8}(\.(\d){8})*$");
```

Comment:
>Match match = Regex.Match(tagPath, @"\A\d{8}(\.(\d){8})*$"); [](start = 12, length = 60)

If we're going to use this a lot, it is probably best to declare this regex outside of the method's scope and compile it with the other [`ctor`]([https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.-ctor?view=net-5.0#System_Text_RegularExpressions_Regex__ctor_System_String_System_Text_RegularExpressions_RegexOptions](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.-ctor?view=net-5.0#System_Text_RegularExpressions_Regex__ctor_System_String_System_Text_RegularExpressions_RegexOptions)_):

```csharp
private static readonly Regex _expectedTagPath = new Regex(@"\A\d{8}(\.(\d){8})*$", RegexOptions.Compiled);
```

Reply:
How about using DicomTagParser.TryParse? it could filter out more invalid cases before calling into more heavier SQL?

Reply:
I'm currently following the pattern for how we validate Uid for retrieve/query. This does less validation than the TryParse does - it doesn't need to validate that it is a valid tag path!

---
In reply to: [564894958](https://github.com/microsoft/dicom-server/pull/501#discussion_r564894958) [](ancestors = 564894958)